### PR TITLE
Bugfix Dockerfile + new build system

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

**Review by maliasadi — APPROVED** [review]
Thanks @vincentmr! This PR adds the missing command to config pyproject.toml as a follow-up to https://github.com/PennyLaneAI/pennylane-lightning/pull/850 and https://github.com/PennyLaneAI/pennylane-lightning/pull/854. Feel free to merge after testing the docker images. :raised_hands:

**Review by mlxd — APPROVED** [review]

File: docker/Dockerfile [hunk 1]
```diff
@@ -73,14 +73,14 @@ RUN python3 -m venv $VIRTUAL_ENV
...
+RUN pip install --no-cache-dir build cmake ninja pybind11 toml wheel
 
 # Download Lightning release and build lightning-qubit backend
 FROM base-build-python AS build-wheel-lightning-qubit
 WORKDIR /opt/pennylane-lightning
 RUN pip uninstall -y pennylane-lightning
-RUN python scripts/configure_pyproject_toml.py
-RUN python -m build
+RUN python scripts/configure_pyproject_toml.py || true
+RUN python -m build --wheel
```
Comment (AmintorDusko) [inline docker/Dockerfile:83]: So, for my own knowledge, to have backward compatibility using the build module it is sufficient to add the `--wheel` flag, right?

Comment (vincentmr) [reply docker/Dockerfile:83]: Yes.

Comment (AmintorDusko) [reply docker/Dockerfile:83]: Beautiful!

**Pull request merged**